src/cpp/r/RExec.cpp

```diff
@@ -595,14 +595,18 @@ IgnoreInterruptsScope::~IgnoreInterruptsScope()
    }
 }
 
-DisableDebugScope::DisableDebugScope(SEXP env): 
-   rdebug_(0), 
-   env_(nullptr)
+DisableDebugScope::DisableDebugScope(SEXP env)
+   : rdebug_(0), 
+     env_(nullptr)
 {
    // nothing to do if no environment 
-   if (env == nullptr) {
+   if (env == nullptr)
+      return;
+   
+   // nothing to do if disabled
+   bool suppressed = r::options::getOption<bool>("rstudio.errors.suppressed", false, false);
```

Comment:
This makes me a little nervous just because we `DisableDebugScope` everywhere. Is this check extremely lightweight/safe?

Reply:
It _should_ be ... but even so, not worth the risk trade-off at this point in the 1.4-JR release cycle.